lib/tree-view.coffee

```diff
@@ -73,6 +74,14 @@ class TreeView
     @element.style.width = "#{state.width}px" if state.width > 0
     @attach() if state.attached
 
+    @disposables.add @onEntryMoved ({initialPath, newPath}) ->
+      updateEditorsForPath(initialPath, newPath)
+
+    @disposables.add @onEntryDeleted ({path}) ->
+      for editor in atom.workspace.getTextEditors()
+        if editor?.getPath()?.startsWith(path)
+          editor.destroy()
```

Comment:
Yeah, you're right - it definitely does get destroyed. Sounds like a bug to me! I'll check for another issue and if one doesn't exist, create one. Thanks.